include/exadg/matrix_free/matrix_free_data.h

```diff
@@ -36,6 +38,40 @@ template<int dim, typename Number>
 struct MatrixFreeData
 {
 public:
+  /**
+   * Default constructor. Does not care about categorization of cells in case of cell based face
+   * loops.
+   */
+  MatrixFreeData()
+  {
+    data.tasks_parallel_scheme = MatrixFree<dim, Number>::AdditionalData::none;
+  }
+
+  /**
+   * Cnstructor that cares about categorization of cells in case of cell based face loops.
+   */
+  MatrixFreeData(std::shared_ptr<Triangulation<dim> const> const triangulation,
+                 bool const                                      use_cell_based_face_loops)
+  {
+    data.tasks_parallel_scheme = MatrixFree<dim, Number>::AdditionalData::none;
+
+    if(use_cell_based_face_loops)
+    {
+      Categorization::do_cell_based_loops(*triangulation, data);
+    }
+  }
+
+  /**
+   * Append MatrixFreeData by the needs of (another) pde_operator provided as argument to this
+   * function.
+   */
+  template<typename Operator>
+  void
+  append(std::shared_ptr<Operator> pde_operator)
```

Comment:
Why not use the name `insert()` here as in the case of the other functions? As far as I see this function does nothing else than delegating the task to the actual operator, which calls the `insert_*()` functions. Or alternatively, one could call the insert functions `append_*`.

Reply:
I don't see this contradiction so clearly. The difference is the following: For the insert functions, the argument passed to the function is the one that is inserted into the vector of items with the same data type. This is not the case for `MatrixFreeData`, i.e., the call

```
matrix_free_data->append(pde_operator);
```

does not directly insert a PDE operator into `MatrixFreeData`, but rather fills the `MatrixFreeData` object with information provided by the PDE operator. Note also that this function already has a function called `append_mapping_flags()`, and I consider the new function more in this logic than in the insert logic.

Regarding a renaming of the `insert_` functions, I think one would have to first change the implementation of the insert functions, which currently assert that the element to be inserted does not already exist, i.e. wouldn't these functions look different when really implementing an append-logic?

Reply:
I am must admit I prefer how it has been done before without the additional indirection via `MatrixFreeData`. But feel free to ignore the comment.